Flannel using CNI

chadswen opened the pull request:
- Updates Controller Manager/Kubelet with Flannel's required configuration for CNI
- Removes old Flannel installation
- Install CNI enabled Flannel DaemonSet/ConfigMap/CNI bins and config on host

bradbeam commented:
ci check this

chadswen commented:
Added RBAC support and updated a doc that referenced Docker configuration which is no longer needed.

chadswen commented:
Rebased again.

mattymo commented on roles/kubernetes-apps/network_plugin/flannel/tasks/main.yml:
```diff
@@ -0,0 +1,32 @@
+  run_once: true
+  when: rbac_enabled and flannel_rbac_manifest.changed
+
+#FIXME: remove when kubespray uses kubernetes 1.7 due to kubernetes/features#124
```
why do we need to purge? we have no previous install that installed flannel as a daemonset. Lines 6-15 should be removed. We can just add proper field, by setting spec.updateStrategy.type to RollingUpdate

mattymo commented on roles/network_plugin/flannel/templates/cni-flannel.yml.j2:
```diff
@@ -0,0 +1,113 @@
+{% if rbac_enabled %}
+      serviceAccountName: flannel
+{% endif %}
+      containers:
```
Insert this block: 
```
      updateStrategy:
        type: RollingUpdate
```

chadswen replied:
`updateStrategy: type: RollingUpdate` should already be in there. Bottom of the file.

mattymo replied:
I'm not sure if it's in the right spot, but it's easy enough to test later

chadswen commented:
* Updated to v.0.2.0 flannel-cni released yesterday which includes CNI v.0.6.0 binaries
* Added `portmap` plugin support
* Fixed an issue that could occur if br_netfilter is not a module and net.bridge.bridge-nf-call-iptables sysctl was not set.
* Addressed linter warnings

chadswen commented:
Just a note, the `br_netfilter` and `net.bridge.bridge-nf-call-iptables` checks are also applicable for weave. Ideally weave and flannel could share code. In another PR, we might want to combine the approach and configure the module/sysctl before kube-proxy is started, otherwise kube-proxy will still warn on startup that neither is set.
